src/Symfony/Component/TypeInfo/Type/ObjectType.php

```diff
@@ -32,14 +32,14 @@ public function __construct(
     ) {
     }
 
-    public function getBaseType(): BuiltinType|self
+    public function getTypeIdentifier(): TypeIdentifier
     {
-        return $this;
+        return TypeIdentifier::OBJECT;
     }
 
-    public function getTypeIdentifier(): TypeIdentifier
+    public function getBaseType(): BuiltinType|self
```

Comment:
Yes, sorry, it's because I went back and forth. This is still a draft, I'll clean up everything at some point.

This is still experimental and I move methods or purposely break BC and like to keep related methods together, so this may happen again, although I will still try and avoid it, fully get your comment and apologize.